src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -180,61 +216,59 @@ private static void updateTrackingInfo(String statusType,
                                            Instant earliestInTransitTime) {
         String upsUpdate = statusType + " " + statusDescription;
-        inTransaction((conn) -> {
-            try (PreparedStatement stmt = conn.prepareStatement(query)) {
-                stmt.setString(1, upsUpdate);
-                stmt.setString(2, date);
-                stmt.setString(3, trackingId);
-                int r = stmt.executeUpdate();
-                if (r != 2) {//number of subkits
-                    logger.error("Update query for UPS tracking updated " + r + " rows! with tracking/return id: " + trackingId);
-                }
+        try (PreparedStatement stmt = conn.prepareStatement(query)) {
+            stmt.setString(1, upsUpdate);
+            stmt.setString(2, date);
+            stmt.setString(3, trackingId);
+            stmt.setString(4, kit.getExternalOrderNumber());
+            int r = stmt.executeUpdate();
+            if (r != 2) {//number of subkits
+                logger.error("Update query for UPS tracking updated " + r + " rows! with tracking/return id: " + trackingId + " for kit w/ external order number "+ kit.getExternalOrderNumber());
+            }
 
-                if (!isReturn) {
-                    if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
-                        KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(conn,SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
-                        if (kitDDPNotification != null) {
-                            logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
-                            EventUtil.triggerDDP(conn, kitDDPNotification);
-                        }
-                        else {
-                            logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
-                        }
+            if (!isReturn) {
+                if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(conn,SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(conn, kitDDPNotification);
                     }
+                    else {
+                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
```

Comment:
What does nullness here imply at a higher level?  If it implies that a certain thing is broken and needs tending to, please bake that into the comment.

Reply:
@pegahtah can you look into that please

Reply:
@SimoneMaiwald a null ness here means a difference in data between ES and DSM. It some times happen in test environments where test users gets deleted but I don't think we will really face it in prod.

However, based on what we learned maybe we should change the SELECT query to not use `SELECT_BY_TRACKING_NUMBER` since now we can have multiple?